src/emailservice/email_server.py

```diff
@@ -146,4 +147,10 @@ def start(dummy_mode):
 
 if __name__ == '__main__':
   logger.info('starting the email service in dummy mode.')
+   # Start the Stackdriver Profiler Python agent
+  try:
+    googlecloudprofiler.start(service='email_server', service_version='1.0.0', verbose=0)
```

Comment:
@kalyanac when this app runs inside Istio, the networking will not be available during the first few milliseconds. 

So if `googlecloudprofiler.start` doesn't have a **retry** mechanism (that ideally runs in the background), your profiling scenario won't work when it's deployed to Istio.

We have 3 retries with 1-2 sec intervals before calling it failure, in most microservices in this repo.

Reply:
See how we call initProfiling method in another service **in the background** with the `go` keyword

https://github.com/GoogleCloudPlatform/microservices-demo/blob/342318a9aef7514c91c08a61eb42fde414976c29/src/shippingservice/main.go

Similarly, initProfiling() has retries internally:

https://github.com/GoogleCloudPlatform/microservices-demo/blob/342318a9aef7514c91c08a61eb42fde414976c29/src/shippingservice/main.go#L192-L212

Reply:
I added a retry loop.